gcsio/src/main/java/com/google/cloud/hadoop/gcsio/AdaptiveFileAccessPattern.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.gcsio;
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageReadOptions.Fadvise;
+import com.google.common.flogger.GoogleLogger;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.ListIterator;
+
+class AdaptiveFileAccessPattern implements Closeable {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private final StorageResourceId resourceId;
+  private final GoogleCloudStorageReadOptions readOptions;
+  private boolean isPatternOverriden = false;
+  private boolean randomAccess;
+  private long lastServedIndex = -1;
+  // Keeps track of distance between consecutive requests
+  private BoundedList<Long> consecutiveRequestsDistances;
+
+  @Override
+  public void close() throws IOException {
```

Comment:
Close is not doing much. Why we need this to be Closeable?

Reply:
Not required anymore. Removed.